ee/codegen/src/__test__/nodes/conditional-node.test.ts

```diff
@@ -1022,3 +1024,101 @@ describe("Conditional Node with OR combinator generates pipe operator", () => {
     expect(generatedCode).toMatchSnapshot();
   });
 });
+
+describe("Conditional Node with NodePorts defined", () => {
+  let workflowContext: WorkflowContext;
+  let writer: Writer;
+  let node: ConditionalNode;
+
+  beforeEach(async () => {
+    workflowContext = workflowContextFactory();
+    writer = new Writer();
+
+    const nodePortsData: NodePort[] = [
+      nodePortFactory({
+        type: "IF",
+        expression: {
+          type: "UNARY_EXPRESSION",
+          lhs: {
+            type: "CONSTANT_VALUE",
+            value: {
+              type: "STRING",
+              value: "hello",
+            },
+          },
+          operator: "null",
+        },
+      }),
+    ];
+
+    const nodeData: ConditionalNodeType = {
+      id: "b81a4453-7b80-41ea-bd55-c62df8878fd3",
+      type: WorkflowNodeType.CONDITIONAL,
+      data: {
+        label: "Conditional Node",
+        targetHandleId: "842b9dda-7977-47ad-a322-eb15b4c7069d",
+        conditions: [
+          {
+            id: "d93fecd6-0f6c-4c13-b572-2254404650cd",
+            type: "IF",
+            sourceHandleId: "b9a8c7af-68bf-42e0-9643-a62d73734c74",
+            data: {
+              id: "55f430c7-a98e-4c40-a2b3-455f7152556a",
+              rules: [
+                {
+                  id: "895d82e8-e6d4-4033-9ae4-581e8c2dc45c",
+                  fieldNodeInputId: "f804da22-b3d2-43d0-8c6f-321434c28ead",
+                  operator: "null",
+                },
+              ],
+              combinator: "AND",
+            },
+          },
+        ],
+        version: "2",
+      },
+      inputs: [
+        {
+          id: "f804da22-b3d2-43d0-8c6f-321434c28ead",
+          key: "895d82e8-e6d4-4033-9ae4-581e8c2dc45c.field",
+          value: {
+            rules: [
+              {
+                type: "CONSTANT_VALUE",
+                data: {
+                  type: "STRING",
+                  value: "hello",
```

Comment:
[nit] Can we make the two have different values? bc they are both `"hello"`, it's hard to see which one is winning from the test